providers: move trait noops into MetadataProvider

darkmuggle opened the pull request:
Generally in all the providers there are trait(s) that are not
implemented or used by the provider; they all had various methods of
noop'ing. This change unifies the noop handling by implementing the
required fn in the trait definition itself.

Signed-off-by: Ben Howard <ben.howard@redhat.com>

lucab commented on src/providers/mod.rs:
```diff
@@ -195,19 +195,38 @@ fn write_ssh_keys(user: User, ssh_keys: Vec<PublicKey>) -> Result<()> {
-    fn boot_checkin(&self) -> Result<()>;
+    fn attributes(&self) -> Result<HashMap<String, String>> {
+        let attributes = maplit::hashmap! {};
+        Ok(attributes)
```
Nit: `Ok(HashMap::new())` should be enough here (it also avoids a macro-expansion pass).

darkmuggle replied:
Indeed.... my rust noob status is showing :) I had wondered about going with anonymous hashmap there. The tip is appreciated.

Commit:
providers/mod.rs: move trait noops into MetadataProvider

Generally in all the providers there are trait(s) that are not
implemented or used by the provider; they all had various methods of
noop'ing. This change unifies the noop handling by implementing the
required fn in the trait definition itself.

Signed-off-by: Ben Howard <ben.howard@redhat.com>
